simpleviewinc/keg-cli
# docker inspect task

## Goal

* Add a docker inspect keg-cli command

## Updates

* Added a `docker inspect` task
* Added a tap wrapper task to call docker inspect

## Testing

**Setup**
* Pull the pr => `keg pr 77`
* Run a tap that uses the `keg-proxy`, I used retheme =>  `keg retheme start`

**Test 1**
* Run command => `keg d inspect`
  * It should ask what docker item type you want to inspect ( `image` || `container` ) 
  * Select either, and from the selected type if should then ask you which item of the selected type to inspect
    * For example selecting the `container` type,
      * Should ask you which container to inspect, and show a list of all containers 
   * Select an item, and ensure it prints the inspect object
  * Run the command again => `keg d inspect`
  * Then select the other type
    * Ensure it shows a list of items of the other type
    * Select an item from the list, and ensure it prints the inspect object

**Test 2**
* Run command => `keg d inspect --image`
* It should show a list of images
* Select an image, and ensure it prints the inspect object

**Test 3**
* Run command => `keg d inspect --container`
* It should show a list of containers
* Select an container, and ensure it prints the inspect object

**Test 4**
* Run command => `keg retheme docker inspect` || `keg retheme d in`
* It should ask if you want to inspect a container or image
  * Select one, and ensure it prints the inspect object 
* Run command again, but select the other `type`
* Ensure it prints the inspect object for that type

**Test 5**
* Run command => `keg retheme docker inspect --image` || `keg retheme d in -i`
* Ensure it prints the inspect object for the retheme image

**Test 6**
* Run command => `keg retheme docker inspect --container` || `keg retheme d in -c`
* Ensure it prints the inspect object for the retheme container
